src/templates/stepper.html

```diff
@@ -180,59 +197,127 @@ <h4><span class="help float-right"> By clicking the <code>Previous State</code>
                         </div>
                     </div>
                     {% endfor %}
-                </div>
-            </div>
-            <!-- This view lets you look at multiple tree views at once, for state by state. -->
-            <div class="tab-pane fade" id="table-view" role="tabpanel">
-
-
-                <p>
-                Sub-formulae are demarcated by their bounding boxes, which
-                indicate whether they are <span class="satformula">satisfied</span> or
-                                    <span class="unsatformula">unsatisfied</span>.
-                                   
-                </p>
-                <!-- Table view goes here -->
-                <table class="table table-bordered" id="stateTable">
-                    <thead>
-                        <tr>
-                            <th class="w-auto"></th>
-                            {% for ps in prefixstates %}
-                                <th></th>
-                            {% endfor %}
-                            {% if cyclestates %}
-                                <th class="cycle-state-col" colspan="{{ cyclestates|length }}">Cycle</th>
-                            {% endif %}
-                        </tr>
-                    </thead>
-                    <tbody>
-                        <tr>
-                            <th class="w-auto">Trace State</th>
-                            {% for ps in prefixstates %}
-                                <td><pre>{{ ps.formattedTraceAssignment }}</pre></td>
-                            {% endfor %}
-                            {% for ps in cyclestates %}
-                                <td class="cycle-state-col"><pre>{{ ps.formattedTraceAssignment }}</pre></td>
-                            {% endfor %}
-                        </tr>
-                        <tr>
-                            <th class="w-auto">
-                                Formula
-                                </span>
-                            </th>
-                            {% for ps in prefixstates %}
-                                <td>
-                                    {{ ps.formulaAsHTML | safe }}
-                                </td>
-                            {% endfor %}
-                            {% for ps in cyclestates %}
-                                <td class="cycle-state-col">
-                                    {{ ps.formulaAsHTML | safe }}
-                                </td>
-                            {% endfor %}
-                        </tr>
-                    </tbody>
-                </table>
+                </div>
+            </div>
+            <!-- ================================ -->
+            <!-- MATRIX VIEW - Binary matrix display of subformula satisfaction over time -->
```

Comment:
Table View BEFORE matrix view. Table View in Beta. Matrix view in Alpha.

Reply:
nder(app, template, context)
  File "/venv/lib/python3.9/site-packages/flask/templating.py", line 131, in _render
    rv = template.render(context)
  File "/venv/lib/python3.9/site-packages/jinja2/environment.py", line 1295, in render
    self.environment.handle_exception()
  File "/venv/lib/python3.9/site-packages/jinja2/environment.py", line 942, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "/src/templates/stepper.html", line 277, in top-level template code
    {% set num_columns = matrix_data.rows[0].values|length %}
TypeError: object of type 'builtin_function_or_method' has no len()